csrc/net/ncnn/ncnn_net.cpp

```diff
@@ -22,18 +22,39 @@ Result<void> NCNNNet::Init(const Value& args) {
   auto& context = args["context"];
   device_ = context["device"].get<Device>();
   stream_ = context["stream"].get<Stream>();
-
+  if (args.contains("use_vulkan")) {
+    net_.opt.use_vulkan_compute = args["use_vulkan"].get<bool>();
+  }
   if (!device_.is_host()) {
     return Status(eNotSupported);
   }
 
   auto name = args["name"].get<std::string>();
   auto model = context["model"].get<Model>();
   OUTCOME_TRY(auto config, model.GetModelConfig(name));
-
+  auto precision = config.precision;
+  if (precision == "FP16") {
+    net_.opt.use_fp16_packed = true;
+    net_.opt.use_fp16_storage = true;
+    net_.opt.use_fp16_arithmetic = true;
+  } else if (precision == "INT8") {
+    // in android platform, ncnn will automatically start FP16 accelerate.
+    // In INT8 case, we set fp16 as false explicitly.
+    net_.opt.use_int8_packed = true;
+    net_.opt.use_int8_storage = true;
+    net_.opt.use_int8_arithmetic = true;
+    net_.opt.use_fp16_packed = false;
```

Comment:
I think it's not necessary to set `net_.opt.use_fp16_packed`, `net_.opt.use_fp16_storage` and `net_.opt.use_fp16_arithmetic`. Aren't their default value `false`? If so, the `else` branch is not needed either.

Reply:
> I think it's not necessary to set `net_.opt.use_fp16_packed`, `net_.opt.use_fp16_storage` and `net_.opt.use_fp16_arithmetic`. Aren't their default value `false`? If so, the `else` branch is not needed either.

No, please refer to https://github.com/Tencent/ncnn/issues/2885. In android platform, the default value of `use_fp16_arithmetic` is true.